paper-api/src/main/java/io/papermc/paper/event/player/PlayerLoadedEvent.java

```diff
@@ -0,0 +1,62 @@
+package io.papermc.paper.event.player;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Called when a player is set as loaded. This either happens when the player sends the player loaded packet after loading the world (closing the downloading terrain screen) or when the player
+ * has not sent the packet for 60 ticks after joining the server or respawning.
+ */
+@NullMarked
+public class PlayerLoadedEvent extends PlayerEvent {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final Cause cause;
+
+    @ApiStatus.Internal
+    public PlayerLoadedEvent(final @NotNull Player who, final Cause cause) {
```

Comment:
Unnecesary NotNull if the class has the NullMarked annotation

Reply:
Yeah, the NotNull annotation was created by Intellij. Will remove

Reply:
This is not completely resolved for getHandlers and can you move the handler related methods at the end of the class for consistency with other events. Javadocs could be spread on more lines.

Reply:
@Lulu13022002 Thanks, pushed a fix